pkg/virt-api/webhooks/defaults.go

```diff
@@ -168,6 +170,10 @@ func setDefaultResourceRequests(clusterConfig *virtconfig.ClusterConfig, spec *v
 			overcommit := clusterConfig.GetMemoryOvercommit()
 			if overcommit == 100 {
 				resources.Requests[k8sv1.ResourceMemory] = *memory
+			} else if overcommit > 100 && guestMemory && util.IsVFIOVMI(spec) {
```

Comment:
Do we need `guestMemory`? If hugepages are used, we probably also don't want to overcommit, do we?
Also, it looks like we trying to avoid "requested memory > guest memory" here, not "guest memory > requested memory" (PR title)

Reply:
You're right about the logic being wrong here; it should be checking that the resulting value of the overcommit calculation (which will be the new requested memory value) is not less than the guest memory. I am not sure about the hugepages case, @fossedihelm do you know?

Reply:
We should check if with `hugePages.pageSize` > `memory.Requests` the `Process exited prior to exec: libvirt: error : cannot limit locked memory of process` is being triggered. Honestly, I don't know